homeassistant/components/climate/venstar.py

```diff
@@ -0,0 +1,287 @@
+
+    def update(self):
+        """Update the data from the thermostat."""
+        _LOGGER.info("Refreshing data from your Venstar Thermostat.")
+        info_success = self._client.update_info()
+        sensor_success = self._client.update_sensors()
+        if not info_success or not sensor_success:
+            _LOGGER.error("Failed to update data from your Thermostat.")
+
+# Thermostat config
+    @property
+    def supported_features(self):
+        """Return the list of supported features."""
+        if self._client.mode == self._client.MODE_AUTO:
+            return (SUPPORT_TARGET_TEMPERATURE |
+                    SUPPORT_TARGET_TEMPERATURE_HIGH |
+                    SUPPORT_TARGET_TEMPERATURE_LOW |
+                    SUPPORT_TARGET_HUMIDITY |
+                    SUPPORT_FAN_MODE |
+                    SUPPORT_OPERATION_MODE)
+
+        else:
+            return (SUPPORT_TARGET_TEMPERATURE |
+                    SUPPORT_TARGET_HUMIDITY |
+                    SUPPORT_FAN_MODE |
+                    SUPPORT_OPERATION_MODE)
+
+    @property
+    def name(self):
+        """Return the name of the thermostat."""
+        return self._client.name
+
+    @property
+    def precision(self):
+        """Return the precision of the system.
+
+        Venstar temperature values are passed back and forth in the
+        API as whole degrees C or F.
+        """
+        return PRECISION_WHOLE
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement, as defined by the API."""
+        if self._client.tempunits == self._client.TEMPUNITS_F:
+            return TEMP_FAHRENHEIT
+        else:
+            return TEMP_CELSIUS
+
+    @property
+    def fan_list(self):
+        """Return the list of available fan modes."""
+        return self._fan_list
+
+    @property
+    def operation_list(self):
+        """Return the list of available operation modes."""
+        return self._operation_list
+
+# Current Values
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        return self._client.get_indoor_temp()
+
+    @property
+    def current_humidity(self):
+        """Return the current humidity."""
+        return self._client.get_indoor_humidity()
+
+    @property
+    def current_operation(self):
+        """Return current operation ie. heat, cool, idle."""
+        if self._client.mode == self._client.MODE_HEAT:
+            return STATE_HEAT
+        elif self._client.mode == self._client.MODE_COOL:
+            return STATE_COOL
+        elif self._client.mode == self._client.MODE_AUTO:
+            return STATE_AUTO
+        else:
+            return STATE_IDLE
+
+    @property
+    def current_fan_mode(self):
+        """Return the fan setting."""
+        if self._client.fan == self._client.FAN_AUTO:
+            return STATE_AUTO
+        else:
+            return STATE_ON
+
+    @property
+    def state_attributes(self):
```

Comment:
Please don't override this. Instead, implement `device_state_attributes()`. The dict returned from that method will be merged with the `state_attributes` from the climate component.